nixos/modules/system/boot/stratisroot.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+let
+  types = lib.types;
+in
+{
+  options.boot.stratis = {
+    rootPoolUuid = lib.mkOption {
+      type = types.uniq types.str;
+      description = lib.mdoc ''
+        UUID of the stratis pool that the root fs is located in
+      '';
+      example = "04c68063-90a5-4235-b9dd-6180098a20d9";
+    };
+  };
+  config = {
+    assertions = [
+      {
+        assertion = config.boot.initrd.systemd.enable;
+        message = "stratis root fs requires systemd initrd";
+      }
+      {
+        assertion = lib.strings.hasPrefix "/dev/stratis/" config.fileSystems."/".device;
+        message = "root file system must live in /dev/stratis";
+      }
+      {
+        assertion = config.boot.plymouth.enable;
+        message = "plymouth must be enabled for stratis pool password prompt";
+      }
```

Comment:
This assertion is probably not true. If it's using `systemd-ask-password`, then any ask-password agent will work, such as the console one that systemd stage 1 has by default. Can you verify whether or not that's correct?

Reply:
`systemd-ask-password` is only called in `stratis-fstab-setup`, which is not being used in this PR?